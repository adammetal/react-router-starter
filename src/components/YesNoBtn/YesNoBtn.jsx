import styled from "styled-components";

const Button = styled.button`
  outline: none;
  border: 2px solid black;
  border-radius: 3px;
  color: white;
  padding: 0.5rem 2rem;
`;

const YesButton = styled(Button)`
  background-color: green;
`;

const NoButton = styled(Button)`
  background-color: red;
`;

const YesNoBtn = (props) => {
  return (
    <div>
      Do you need that kitten???
      <div>
        <YesButton onClick={props.onYes}>Yes</YesButton>
        <NoButton onClick={props.onNo}>No</NoButton>
      </div>
    </div>
  );
};

export default YesNoBtn;
