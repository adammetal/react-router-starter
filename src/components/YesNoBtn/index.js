import YesNoBtn from "./YesNoBtn";

export default YesNoBtn;
