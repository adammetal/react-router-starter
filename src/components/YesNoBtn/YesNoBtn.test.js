import React from 'react';
import { fireEvent, render, screen } from "@testing-library/react";
import YesNoBtn from "./YesNoBtn";

test("YesNoButton Snapshot test", () => {
  const { container } = render(<YesNoBtn onYes={() => {}} onNo={() => {}} />);
  expect(container).toMatchSnapshot();
});

test('YesNoButton #onYes event', () => {
  const onYes = jest.fn();
  const onNo = jest.fn();

  render(<YesNoBtn onYes={onYes} onNo={onNo} />);

  const yesBtn = screen.getByText('Yes');
  fireEvent.click(yesBtn);

  expect(onYes).toBeCalled();
});
