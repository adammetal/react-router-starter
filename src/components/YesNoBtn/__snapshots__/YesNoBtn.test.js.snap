// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YesNoButton Snapshot test 1`] = `
<div>
  <div>
    Do you need that kitten???
    <div>
      <button
        class="sc-bczRLJ sc-gsnTZi cHRURN deEhbE"
      >
        Yes
      </button>
      <button
        class="sc-bczRLJ sc-dkzDqf cHRURN kekkCb"
      >
        No
      </button>
    </div>
  </div>
</div>
`;
